### PR TITLE
SAK-42414 allow a gradebook category to equally weight all items in category

### PR DESCRIPTION
**Pull request by ottenhoff**
https://jira.sakaiproject.org/browse/SAK-42414

* Can set a category to be equal weight
* Can drop lowest/highest/etc in an equal weight category
  * Can drop lowest despite points discrepancies (because cat is equal)
* Can have an equal weight + extra credit category

### CODE REVIEW

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/SettingsCategoryPanel.java  2
- gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/AddOrEditGradeItemPanel.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/sakai/sakai/pullRequest?prid=4611406&bid=15470063)

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/AddOrEditGradeItemPanel.java  1
- gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/SettingsCategoryPanel.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/sakai/sakai/pullRequest?prid=4611406&bid=15470063)

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/SettingsCategoryPanel.java  2
- gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/AddOrEditGradeItemPanel.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/sakai/sakai/pullRequest?prid=4611406&bid=15470063)

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/SettingsCategoryPanel.java  3
- gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/AddOrEditGradeItemPanel.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/sakaiproject/sakai/pullRequest?prid=4611406&bid=15470063)

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/SettingsCategoryPanel.java  3
- gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/AddOrEditGradeItemPanel.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/sakaiproject/sakai/pullRequest?prid=4611406&bid=15470063)

**Comment by fostersdesign** [comment]
https://jira.sakaiproject.org/browse/SAK-42414

File: edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java [hunk 1]
```diff
@@ -1056,11 +1060,25 @@ private List getTotalPointsEarnedInternal(final String studentId, final Gradeboo
...
 										cateScoreMap.put(cate.getId(), pointsEarned);
 									}
+
+									// Running average for an equal-weight category
+/*									final BigDecimal singleItemAverage = pointsEarned.divide(pointsPossible, GradebookService.MATH_CONTEXT);
+									if (cateMeanScoreMap.get(cate.getId()) != null) {
+										cateMeanScoreMap.put(cate.getId(), cateMeanScoreMap.get(cate.getId()).add(singleItemAverage, GradebookService.MATH_CONTEXT));
+									} else {
+										cateMeanScoreMap.put(cate.getId(), pointsEarned);
+									}*/
```
Comment (bjones86) [inline edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java]: Can we just remove this block that's commented out?

Comment (ottenhoff) [reply edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java]: yes agree

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/AddOrEditGradeItemPanel.java  1
- gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/SettingsCategoryPanel.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/sakaiproject/sakai/pullRequest?prid=4611406&bid=15470063)

**Comment by ottenhoff** [comment]
CC: @davidpbauer

**Pull request merged**